src/agentic/generate_commit_message.rs

```diff
@@ -0,0 +1,297 @@
+        vec![
+            ChatMessage {
+                role: "system".to_string(),
+                content: ChatContent::SimpleText(DIFF_WITH_USERS_TEXT_PROMPT.to_string()),
+                ..Default::default()
+            },
+            ChatMessage {
+                role: "user".to_string(),
+                content: ChatContent::SimpleText(format!(
+                    "Commit message:\n```\n{}\n```\nDiff:\n```\n{}\n```\n",
+                    text, diff
+                )),
+                ..Default::default()
+            },
+        ]
+    } else {
+        vec![
+            ChatMessage {
+                role: "system".to_string(),
+                content: ChatContent::SimpleText(DIFF_ONLY_PROMPT.to_string()),
+                ..Default::default()
+            },
+            ChatMessage {
+                role: "user".to_string(),
+                content: ChatContent::SimpleText(format!("Diff:\n```\n{}\n```\n", diff)),
+                ..Default::default()
+            },
+        ]
+    };
+    let model_name = match try_load_caps_quickly_if_not_present(gcx.clone(), 0).await {
+        Ok(caps) => caps
+            .read()
+            .map(|x| Ok(x.code_chat_default_model.clone()))
+            .map_err(|_| "Caps are not available".to_string())?,
+        Err(_) => Err("No caps available".to_string()),
+    }?;
+    let ccx: Arc<AMutex<AtCommandsContext>> = Arc::new(AMutex::new(
+        AtCommandsContext::new(
+            gcx.clone(),
+            N_CTX,
+            1,
+            false,
+            messages.clone(),
+            "".to_string(),
+            false,
+        )
+            .await,
+    ));
+    let new_messages = subchat_single(
+        ccx.clone(),
+        model_name.as_str(),
+        messages,
+        vec![],
+        None,
+        false,
+        Some(TEMPERATURE),
+        None,
+        1,
+        None,
+        None,
+        None,
+    )
+        .await
+        .map_err(|e| format!("Error: {}", e))?;
+
+    let commit_message = new_messages
+        .into_iter()
+        .next()
+        .map(|x| {
+            x.into_iter().last().map(|last_m| match last_m.content {
+                ChatContent::SimpleText(text) => Some(text),
+                ChatContent::Multimodal(_) => None,
+            })
+        })
+        .flatten()
+        .flatten()
+        .ok_or("No commit message was generated".to_string())?;
+    Ok(remove_fencing(&commit_message))
+}
+
+pub async fn _generate_commit_message_for_projects(
+    gcx: Arc<ARwLock<GlobalContext>>,
+) -> Result<HashMap<PathBuf, String>, String> {
+    let project_folders = gcx.read().await.documents_state.workspace_folders.lock().unwrap().clone();
+    let mut commit_messages = HashMap::new();
+
+    for folder in project_folders {
+        let command = if let Some((_, vcs_type)) = detect_vcs_for_a_file_path(&folder).await {
+            match vcs_type {
+                "git" => "git diff",
```

Comment:
I was changing where we use git cli to change it to use git2 crate (we already had it, and it works without cli, so it can work also on docker isolation without installing git cli, for now this is fine, but I have a method called `git_diff_from_all_changes` that could be helpful for this, now it includes unstaged changes but it can be modified to support both options, I will probably do that later